libraries/AP_NavEKF3/derivation/main.py

```diff
@@ -1,4 +1,10 @@
-#!/usr/bin/env python3
+#!/usr/bin/env nix-shell
+#! nix-shell --pure -i python3 -p "python3.withPackages (p: [ p.numpy p.sympy ])"
+#! nix-shell -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/70bdadeb94ffc8806c0570eb5c2695ad29f0e421.tar.gz
+
+# above pins Python 3.10.13, Numpy 1.24.2, and Sympy 1.11.1 (and deps)
+# using the last nixos-23.05 branch commit
+
```

Comment:
This means you can just run the `generate.sh` script and the environment will be automatically set up before the script starts and torn down once it ends. Now that I think about it I should be able to move the nix-shell to the shell script to avoid disrupting non-Nix users.